packages/host/src/components/hooks.ts

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useState } from 'react';
+import create from 'zustand';
+import { AppRoute } from '@openmsupply-client/config';
+import {
+  AuthenticationError,
+  AuthError,
+  LocalStorage,
+  ServerStatus,
+  useNavigate,
+} from '@openmsupply-client/common';
+import { useHost } from '../api/hooks';
+
+const SERVER_RESTART_TIMEOUT = 90000;
+interface InitialiseForm {
+  error?: AuthenticationError;
+  isLoading: boolean;
+  password: string;
+  siteId?: number;
+  username: string;
+  url: string;
+  setError: (error?: AuthenticationError) => void;
+  setIsLoading: (isLoading: boolean) => void;
+  setPassword: (password: string) => void;
+  setUsername: (username: string) => void;
+  setSiteId: (siteId: number) => void;
+  setUrl: (url: string) => void;
+}
+
+const useInitialiseFormState = create<InitialiseForm>(set => ({
+  error: undefined,
+  isLoading: false,
+  password: '',
+  username: '',
+  url: '',
+  siteId: undefined,
+  setError: (error?: AuthenticationError) =>
+    set(state => ({ ...state, error })),
+  setIsLoading: (isLoading: boolean) => set(state => ({ ...state, isLoading })),
+  setPassword: (password: string) => set(state => ({ ...state, password })),
+  setUsername: (username: string) => set(state => ({ ...state, username })),
+  setUrl: (url: string) => set(state => ({ ...state, url })),
+  setSiteId: (siteId: number) => set(state => ({ ...state, siteId })),
+}));
+
+export const useInitialiseForm = () => {
+  const state = useInitialiseFormState();
+  const navigate = useNavigate();
+  const { mutateAsync: restart } = useHost.utils.restart();
+  const {
+    setIsLoading,
+    password,
+    setPassword,
+    username,
+    error,
+    setError,
+    url,
+    siteId,
+  } = state;
+  const [isPolling, setIsPolling] = useState(false);
+  const [isBootstrap, setIsBootstrap] = useState(false);
+  const { mutateAsync: update } = useHost.sync.update();
+  const { data } = useHost.utils.settings({
+    refetchInterval: 5000,
```

Comment:
the `refetchInterval` will have the query requery every 5s. Originally had it as 1s but worried that the hits to the server might have slowed the restart and it was taking over a minute anyway, so a few extra seconds didn't matter. Wonder if it should be reduced again though.